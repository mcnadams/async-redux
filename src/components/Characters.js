import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';

function Characters({ characters }) {
  const characterList = characters.map(character => {
    return (
      <li key={character._id} style={{ 
        'listStyleType': 'none',
        'width': '25vw',
        'margin': '5px'
      }}>
        <Character character={character} />
      </li>
    );
  });

  return (
    <ul style={{ 
      'display': 'flex',
      'flexWrap' : 'wrap'
    }}>
      {characterList}
    </ul>
  );

}

Characters.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    photoUrl: PropTypes.string.isRequired,
    _id: PropTypes.string.isRequired
  }))
};

export default Characters; 
