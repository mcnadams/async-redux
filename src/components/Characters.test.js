import { shallow } from 'enzyme';
import React from 'react';
import Characters from './Characters';

describe('Characters component', () => {
  it('renders Characters', () => {
    const wrapper = shallow(<Characters characters={[{ 
      name: 'Bonnie',
      photoUrl: './path/to/image',
      id: '2'
    }]}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
