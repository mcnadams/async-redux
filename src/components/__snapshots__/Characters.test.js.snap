// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters component renders Characters 1`] = `
<ul
  style={
    Object {
      "display": "flex",
      "flexWrap": "wrap",
    }
  }
>
  <li
    style={
      Object {
        "listStyleType": "none",
        "margin": "5px",
        "width": "25vw",
      }
    }
  >
    <Character
      character={
        Object {
          "id": "2",
          "name": "Bonnie",
          "photoUrl": "./path/to/image",
        }
      }
    />
  </li>
</ul>
`;
